src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -328,17 +328,24 @@ def _cleanup_gaps(self) -> None:
         self._gaps = sorted(self._gaps, key=lambda x: x.start.timestamp())
 
         i = 0
-        while i < len(self._gaps) - 1:
+        while i < len(self._gaps):
```

Comment:
Def should have a test for this and the other new case

Reply:
There is a case for the new case at least

Reply:
Added another test case for the "old" bug.